### PR TITLE
watcher and ntfy rework with spell correction

### PR DESCRIPTION
**Pull request by mhamidjamil**
Correct some spellings mistakes, ntfy service added with guidelines, watcher implemented but need to adjust delay so it works fine or might be go with multi-threading concept to overcome this issue.

### CODE REVIEW

File: serial_communication/web/app.py [hunk 1]
```diff
@@ -153,21 +164,25 @@ def one_time_task():
                         str(random.randint(10000, 99999)))
                     BOOT_MESSAGE_SEND = True
                 time.sleep(5)
-            say_to_serial("sms sending?")
-            time.sleep(15)
-            send_ngrok_link()
+                say_to_serial("sms sending?")
+                time.sleep(15)
+                send_ngrok_link()
```
Comment (wasey-rao) [inline serial_communication/web/app.py:169]: I guess e have done this before please update your branches head

File: serial_communication/web/app.py [hunk 2]
```diff
@@ -153,21 +164,25 @@ def one_time_task():
...
     except Exception as ott: # pylint: disable=broad-except
         exception_logger("one_time_task", ott)
 
 if __name__ == '__main__':
     try:
         # lsof -i :6677 #to know which process is using this port
         current_time = datetime.now()
+        initialize_port()
+        flash()
+        beep()
```
Comment (wasey-rao) [inline serial_communication/web/app.py:179]: I don't thing they work fine do they?

File: serial_communication/web/routes/watcher.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    send_to_serial_port("buzzer on")
+
+def buzzer_off():
+    "to turn Buzzer off"
+    send_to_serial_port("buzzer off")
+
+def beep():
+    """to beep"""
+    buzzer_on()
+    # time.sleep(2)
```
Comment (wasey-rao) [inline serial_communication/web/routes/watcher.py:79]: Please add delay and i guess you should go with threading as i mentioned earlier app.py just skip this part it just call the function and then get back to here it left so at least uncomment delays.

File: serial_communication/web/app.py [hunk 4]
```diff
@@ -31,43 +36,47 @@ def get_serial_ports():
...
+@app.route('/update_port', methods=['POST'])
+def update_port():
+    """To update port dynamically from the web page"""
+    try:
+        selected_port = request.get_json().get('port')
+        if selected_port is not None:
+            print("Trying to change to: " + selected_port)
+            global ser, COMMUNICATION_PORT # pylint: disable=global-statement
+            ser = serial.Serial(selected_port, 115200)
+            COMMUNICATION_PORT = ser
```
Comment (wasey-rao) [inline serial_communication/web/app.py:70]: @mhamidjamil we got problem here, if you change this port then you communication will be disturbed with ttgo which we don't want to, so please update the port in such a way so the communication with the ttgo module should also not disturb.

**Review by wasey-rao — CHANGES_REQUESTED** [review]
Please resolve comments and update you branch heads

File: serial_communication/web/app.py (hunk 4 above)
Comment (mhamidjamil) [reply serial_communication/web/app.py:70]: You are write like if we forget to set back the TTGO port then the communication will be unexpected.

Comment (mhamidjamil) [reply serial_communication/web/app.py:70]: I have verified serial_handler is just responsible for communication with ttgo so no need to worry about ttgo communication i also have verified it from my side by changing port from web and then leave the page after rebooting ttgo the prayer time was updated after 5 minutes.

File: serial_communication/web/app.py (hunk 2 above)
Comment (mhamidjamil) [reply serial_communication/web/app.py:179]: yes, you are right, I just currently make our app able to communicate with watcher.

**Comment by wasey-rao** [comment]
ok then merging it with cross fingers 😉

**Pull request merged**